### PR TITLE
feat: reduce ckb transactions export as csv n+1 querys

### PR DESCRIPTION
**Pull request by rabbitz**

### CODE REVIEW

**Comment by zmcNotafraid** [comment]
Needs to add test.

**Comment by rabbitz** [comment]
https://github.com/nervosnetwork/ckb-explorer/blob/develop/test/test_helper.rb#L176

The timestamp parameter in the `fake_node_block` method is the millisecond value `1557482351075`, but https://github.com/nervosnetwork/ckb-explorer/blob/develop/test/factories/block.rb#L9 The timestamp of the generated block is seconds, `Block.recent.first` Always take the first record, and the corresponding mining_infos value is empty, resulting in an error in https://github.com/nervosnetwork/ckb-explorer/blob/develop/app/jobs/revert_block_job.rb#L122

@zmcNotafraid @ShiningRay

**Comment by ShiningRay** [comment]
I suggest extract the CSV generation code into a separate job class.
So we can reuse them later, or refactor to support background generation.

File: test/factories/block.rb [hunk 1]
```diff
@@ -5,8 +5,7 @@
     number { 10 }
     parent_hash { "0xcba2d1a70602a1def80efbd59629c37a9d6c36f9de7a8ed6d1ca4f76389365e1" }
     nonce { 1757392074788233522 }
-    # FIXME: the timestamp in blocks table is in milliseconds but here is in seconds
-    timestamp { Faker::Time.between(from: 2.days.ago, to: Date.today).to_i }
+    timestamp { Faker::Time.between(from: 2.days.ago, to: Date.today).to_i * 1000 }
```
Comment (zmcNotafraid) [inline test/factories/block.rb:8]: Here should use a fixed value. You can get number 10 block's timestamp from vcr files. It's 1570197479041.

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

**Review by zmcNotafraid — APPROVED** [review]

File: app/jobs/export_address_transactions_job.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    if args[:start_number].present?
+      tx_ids = tx_ids.where("ckb_transactions.block_number >= ?", args[:start_number])
+    end
+
+    if args[:end_number].present?
+      tx_ids = tx_ids.where("ckb_transactions.block_number <= ?", args[:end_number])
+    end
+
+    Rails.cache.realize(args.values, race_condition_ttl: 1.minute) do
```
Comment (ShiningRay) [inline app/jobs/export_address_transactions_job.rb:26]: don't use `Rails.cache.realize`

**Review by ShiningRay — CHANGES_REQUESTED** [review]

**Review by ShiningRay — APPROVED** [review]

**Pull request merged**